Client/Frontend/Browser/ReaddleSparkIntegration.swift

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+struct ReaddleSparkIntegration {
+    static let supportedHeaders = [
+        "recipient",
+        "html",
+        "subject",
+        "textbody"
+    ]
+
+    static func newEmailURLFromMetadata(metadata: MailToMetadata) -> NSURL {
+        var readdleSparkMailURL = "readdle-spark://compose?"
+
+        // The web is a crazy place and some people like to capitalize the hname values in their mailto: links.
+        // Make sure we lowercase anything we found in the metadata since Outlook requires them to be lower case.
+        var lowercasedHeaders = [String: String]()
+        metadata.headers.forEach { (hname, hvalue) in
+            lowercasedHeaders[hname.lowercaseString] = hvalue
+        }
+
+        // If we have both a [ to ] parameter and an hname 'to', combine them according to the RFC.
+        var toParam: String
+        if let toHValue = lowercasedHeaders["to"] {
+            let value = metadata.to.isEmpty ? toHValue : [metadata.to, toHValue].joinWithSeparator("%2C%20")
+            lowercasedHeaders.removeValueForKey("to")
+            toParam = "recipient=\(value)"
+        } else {
+            toParam = "recipient=\(metadata.to)"
+        }
+
+        let queryParams = lowercasedHeaders.filter { (hname, _) in
+            return supportedHeaders.contains(hname)
+            } .map { "\($0)=\($1)" } .joinWithSeparator("&")
+
+
+        readdleSparkMailURL +=
+            queryParams.isEmpty ? toParam : [toParam, queryParams].joinWithSeparator("&")
+        
+        return readdleSparkMailURL.asURL!
```

Comment:
This makes me a bit worried because we are taking values the web and building a URL. Not sure if we can guarantee that this will be a valid URL everytime.